backend/src/test/java/org/cryptomator/hub/api/VaultResourceTest.java

```diff
@@ -723,17 +748,97 @@ public void testListSomeVaultsNoParams() {
+						VALUES
+						('7E57C0DE-0000-4000-8000-FFFFFFFF1111', 'Vault U', 'Vault to update.',
+						 '2020-02-20T20:20:20Z', 'saltU', 42, 'masterkeyU', 'authPubKeyU', 'authPrvKeyU', FALSE);
+						""");
+
+			}
+		}
+
+		@Test
+		@DisplayName("PUT /vaults/7E57C0DE-0000-4000-8000-FFFFFFFF2222 returns 201")
+		public void testCreateVault1() {
+			var uuid = UUID.fromString("7E57C0DE-0000-4000-8000-FFFFFFFF2222");
+			var vaultDto = new VaultResource.VaultDto(uuid, "Test Vault", "Vault to create", Instant.parse("2112-12-21T21:12:21Z"), "masterkeyC", 42, "saltC", "authPubKeyC", "authPrvKeyC", false);
+			given().contentType(ContentType.JSON).body(vaultDto)
+					.when().put("/vaults/{vaultId}", "7E57C0DE-0000-4000-8000-FFFFFFFF2222")
+					.then().statusCode(201)
+					.body("id", equalToIgnoringCase("7E57C0DE-0000-4000-8000-FFFFFFFF2222"))
+					.body("name", equalTo("Test Vault"))
+					.body("description", equalTo("Vault to create"))
+					.body("masterkey", equalTo("masterkeyC"))
+					.body("salt", equalTo("saltC"))
+					.body("iterations", equalTo(42))
+					.body("authPublicKey", equalTo("authPubKeyC"))
+					.body("authPrivateKey", equalTo("authPrvKeyC"))
+					.body("archived", equalTo(false));
+		}
+
+		@Test
+		@DisplayName("PUT /vaults/7E57C0DE-0000-4000-8000-BADBADBADBAD returns 400")
+		public void testCreateVault2() {
+			given().contentType(ContentType.JSON)
+					.when().put("/vaults/{vaultId}", "7E57C0DE-0000-4000-8000-BADBADBADBAD")
+					.then().statusCode(400);
+		}
 
+		@Test
+		@DisplayName("PUT /vaults/7E57C0DE-0000-4000-8000-FFFFFFFF1111 returns 200 with only updated name, description and archive flag")
+		public void testUpdateVault() {
+			var uuid = UUID.fromString("7E57C0DE-0000-4000-8000-FFFFFFFF1111");
+			var vaultDto = new VaultResource.VaultDto(uuid, "VaultUpdated", "Vault updated.", Instant.parse("2112-12-21T21:12:21Z"), "someVaule", -1, "someVaule", "someValue", "someValue", true);
+			given().contentType(ContentType.JSON)
+					.body(vaultDto)
+					.when().put("/vaults/{vaultId}", "7E57C0DE-0000-4000-8000-FFFFFFFF1111")
+					.then().statusCode(200)
+					.body("id", equalToIgnoringCase("7E57C0DE-0000-4000-8000-FFFFFFFF1111"))
+					.body("name", equalTo("VaultUpdated"))
+					.body("description", equalTo("Vault updated."))
+					.body("creationTime", equalTo("2020-02-20T20:20:20Z"))
+					.body("masterkey", equalTo("masterkeyU"))
+					.body("salt", equalTo("saltU"))
+					.body("iterations", equalTo(42))
+					.body("authPublicKey", equalTo("authPubKeyU"))
+					.body("authPrivateKey", equalTo("authPrvKeyU"))
+					.body("archived", equalTo(true));
+		}
+
+		@AfterAll
+		public void deleteData() throws SQLException {
+			try (var s = dataSource.getConnection().createStatement()) {
+				s.execute("""
+						DELETE FROM "vault"
+						WHERE "id" IN ('7E57C0DE-0000-4000-8000-FFFFFFFF1111','7E57C0DE-0000-4000-8000-FFFFFFFF2222');
+						""");
+			}
+		}
```

Comment:
0001 id namespace

Reply:
Fixed in https://github.com/cryptomator/hub/pull/202/commits/1331ecc9215fbfaa25d9797188c9c3bd72db072d